dataland-frontend/src/components/general/ApiKeyCard.vue

```diff
@@ -0,0 +1,99 @@
+<template>
+  <PrimeDialog id="revokeModal" :modal="true" header="Header" footer="Footer" v-model:visible="viewDeleteConfirmation">
+    <template #header>
+      <h2 class="m-0">Delete API Key</h2>
+    </template>
+    Are you sure you want to delete this API key?
+    <strong>If you confirm, your previous token will be invalidated and your applications will stop working.</strong>
+    <template #footer>
+      <PrimeButton label="CANCEL" @click="deleteConfirmToggle" class="p-button-outlined text-sm" />
+      <PrimeButton id="confirmRevokeButton" label="CONFIRM" @click="$emit('revokeKey')" class="text-sm" />
+    </template>
+  </PrimeDialog>
+
+  <div>
+    <div id="existingApiKeyCard" class="surface-card shadow-1 p-3 border-round-sm border-round">
+      <div class="flex justify-content-between mb-3">
+        <div>
+          <div class="text-900 font-medium text-xl text-left">API Key info</div>
+          <span :class="{ 'text-red-700': !isKeyExpired }" class="block text-600 mb-3 mt-6">
+            {{
+              isKeyExpired
+                ? `The API Key will expire on ${formatExpiryDate(expiryDateInDays)}`
+                : `The API Key expired ${formatExpiryDate(expiryDateInDays)}`
+            }}
+          </span>
+        </div>
+
+        <div data-test="userRoles" v-if="userRoles.length">
```

Comment:
We actually always expect to have userRoles. Do you think the v-if here is necessary?

Reply:
ok